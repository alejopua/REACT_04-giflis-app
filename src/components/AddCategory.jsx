import { useState } from "react"

// eslint-disable-next-line react/prop-types
export const AddCategory = ({setCategories}) => {
  const [newValue, setNewValue] = useState('')

  const onSubmit = (e) =>{
    e.preventDefault();
    newValue !== '' ? setCategories((cat) => [...cat, newValue]) : null;
    setNewValue('')
  }

  return (
    <>
      <form onSubmit={onSubmit}>
        <input 
          type="text"
          value={newValue} 
          placeholder="Insert new giflis" 
          onChange={(e) => setNewValue(e.target.value)}
        />
      </form>
    </>
  )
}
