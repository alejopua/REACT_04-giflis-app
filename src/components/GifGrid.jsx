/* eslint-disable react-hooks/exhaustive-deps */
import PropTypes from 'prop-types';
import { getGifs } from '../helpers/getGifs';
import { useState, useEffect } from 'react';

export const GifGrid = ({category}) => {
  const [images, setImages] = useState([])

  const getImages = async() => {
    const newImages = await getGifs(category);
    setImages(newImages);
  }

  useEffect(() => {
    getImages();
  }, []);
  

  return (
    <>
      <h2>{category}</h2>
      <ol>
        {
          images.map( ({id, title}) => (
            <li key={id}>{title}</li>
          ))
        }
      </ol>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};
