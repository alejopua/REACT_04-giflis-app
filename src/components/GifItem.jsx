import PropTypes from 'prop-types'

export const GifItem = ({url, title}) => {
  const newTitle = title.split(' ').splice(0,2).join(' ');

  return (
    <div className='card'>
      <img src={url} alt={title} />
      <p>{newTitle}</p>
    </div>
  )
}

GifItem.propTypes = {
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}