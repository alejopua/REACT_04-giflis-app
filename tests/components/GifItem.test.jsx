import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Testing in GifItem.jsx', () => { 
  const title = 'A title';
  const url   = 'https://...';

  test('Must match the snapshot', () => {
    const { container } = render( <GifItem title={ title } url={ url } /> ) 

    expect( container ).toMatchSnapshot();
  });

  test('must show the image with the title and url', () => {
    render( <GifItem title={ title } url={ url } /> );
    // screen.debug();

    // expect( screen.getByRole('img').alt ).toBe( title );

    const { alt, src } = screen.getByRole('img');
    expect( alt ).toBe( title );
    expect( src ).toBe( url + '/');
  });

  test('must show the title in the component', () => {
    render( <GifItem title={ title } url={ url } /> );

    expect( screen.getByText( title ) ).toBeTruthy();
  });

})