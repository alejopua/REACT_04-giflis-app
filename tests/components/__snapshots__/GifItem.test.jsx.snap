// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in GifItem.jsx Must match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="A title"
      src="https://..."
    />
    <p>
      A title
    </p>
  </div>
</div>
`;
